web/src/test/java/org/openmrs/web/controller/form/FormResourceControllerTest.java

```diff
@@ -0,0 +1,92 @@
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.controller.form;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.openmrs.Form;
+import org.openmrs.FormResource;
+import org.openmrs.customdatatype.datatype.LongFreeTextDatatype;
+import org.openmrs.web.attribute.handler.LongFreeTextFileUploadHandler;
+import org.openmrs.web.test.BaseWebContextSensitiveTest;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.mock.web.MockMultipartHttpServletRequest;
+import org.springframework.util.FileCopyUtils;
+import org.springframework.validation.Errors;
+
+public class FormResourceControllerTest extends BaseWebContextSensitiveTest {
+// 
+//    @Test
+//    public void handleFormResource_shouldStayOnSamePageWhenUnableToSaveFormResource() throws Exception {
+//    	
+//		final String EXPECTED_FORM_ID = "1";
+//		final String EXPECTED_DATATYPE_CLASS_NAME = (new LongFreeTextDatatype()).getClass().getName();
+//		final String EXPECTED_HANDLER_CLASS_NAME = (new LongFreeTextFileUploadHandler()).getClass().getName();
+//		final String EXPECTED_URL = "redirect:addFormResource.form?formId="
+//			+ EXPECTED_FORM_ID + "&datatype=" 
+//			+ EXPECTED_DATATYPE_CLASS_NAME + "&handler="
+//			+ EXPECTED_HANDLER_CLASS_NAME;
+//	    
+//		Errors errors = null;
+//		
+//		Form form = null;
+//		
+//		FormResource resource = null;
+//		
+//		MockMultipartHttpServletRequest request = new MockMultipartHttpServletRequest();
+//		request.setMethod("POST");
+//		request.setParameter("formId", "1");
+//		request.setParameter("datatype", EXPECTED_DATATYPE_CLASS_NAME);
+//		request.setParameter("handler", EXPECTED_HANDLER_CLASS_NAME);
+//	
+//		FormResourceController controller = new FormResourceController();
+//		
+//		String actualUrl = controller.handleAddFormResource(resource, errors, request);
+//		
+//		Assert.assertEquals(EXPECTED_URL, actualUrl);
+//    }
+//    
+    @Test
+    public void handleFormResource_shouldRedirectAfterSavingFormResource() throws Exception {
+    	Form form = new Form();
+    	form.setId(1);
+    	form.setName("test form");
+    	form.setFormId(1);
+    	form.setVersion("1.0.0");
+    	
+    	FormResource resource = new FormResource();
+    	resource.setName("test resource");
+    	resource.setForm(form);
+    	resource.setFormResourceId(1);
+    	resource.setId(1);
+    	
+    	byte[] fileContent = "Hello World".getBytes();
+    	String fileName = "/resources/testfile.ico";
+    	MockMultipartHttpServletRequest request = new MockMultipartHttpServletRequest();
+    	MockMultipartFile file = new MockMultipartFile("resourceValue", fileName, "text/plain", fileContent);
+    	
+    	request.setMethod("POST");
+    	request.setParameter("datatype", (new LongFreeTextDatatype()).getClass().getName());
+    	request.setParameter("handler", (new LongFreeTextFileUploadHandler()).getClass().getName());
+    	request.addFile(file);
+    	
```

Comment:
@dkayiwa no matter what value i pass to the request and the form resource, can't get it to throw the expected exceptions in test. What am saying is, on the page where the form resource is been saved, if you submit no or invalid file, you get an error immediately due to the exceptions thrown. But when writing the unit tests, no matter what file i pass to the multipart request, no exception is thrown.  Also in the FormServiceImpl.saveFormResource() method, no matter what values i contstruct the resource with, it always seem to complete successfully. I don't know why.